src/nbl/ui/CWindowXCB.cpp

```diff
@@ -0,0 +1,296 @@
+#ifdef _NBL_PLATFORM_LINUX_
+
+#include "nbl/core/decl/smart_refctd_ptr.h"
+#include "nbl/core/string/StringLiteral.h"
+
+#include "nbl/system/DefaultFuncPtrLoader.h"
+
+#include "nbl/ui/IWindowXCB.h"
+#include "nbl/ui/CWindowXCB.h"
+#include "nbl/ui/CCursorControlXCB.h"
+#include "nbl/ui/CClipboardManagerXCB.h"
+#include "nbl/ui/CWindowManagerXCB.h"
+
+#include <cstdint>
+#include <string>
+#include <array>
+#include <string_view>
+#include <variant>
+
+namespace nbl::ui {
+
+static bool checkXcbCookie(const Xcb& functionTable, xcb_connection_t* connection, xcb_void_cookie_t cookie) {
+    if (xcb_generic_error_t* error = functionTable.pxcb_request_check(connection, cookie))
+    {
+        printf("XCB error: %d", error->error_code);
+        return false;
+    }
+    return true;
+}
+
+CWindowXCB::CDispatchThread::CDispatchThread(CWindowXCB& window):
+    m_window(window) {
+}
+
+void CWindowXCB::CDispatchThread::work(lock_t& lock){
+    if(m_quit) {
+        return; 
+    }
+    auto& xcb = m_window.m_windowManager->getXcbFunctionTable();
+    auto& connection = m_window.m_connection;
+
+    auto MW_DELETE_WINDOW = connection->resolveAtom(m_window.m_WM_DELETE_WINDOW);
+    auto NET_WM_PING = connection->resolveAtom(m_window.m_NET_WM_PING);
+
+    if(auto event = xcb.pxcb_wait_for_event(connection->getRawConnection())) {
```

Comment:
this puts the thread to sleep if there's no events, correct?

Reply:
yep, xcb implements it own dispatch queue.